snakemake/executors.py

```diff
@@ -387,7 +388,7 @@ def job_args_and_prepare(self, job):
             job.wildcards,
             job.threads,
             job.resources,
-            job.log._plainstrings(),
+            job.log,
```

Comment:
I guess that by mapping the file collections to their `_plainstrings` equivalents, you're trying to obscure the AnnotatedString protocol from  the executors, but I left it this way in hopes at least of getting the functionality ready, and work out the object model with you.

Is there an alternative way to pass on awareness of the stdout/stderr flags to the shell class?

Also ... should this functionality extend to the python `script` blocks?  I can follow up with that, if you'd give me some instruction.